pkg/dataconn/client.go

```diff
@@ -152,10 +115,17 @@ func (c *Client) Close() {
 }
 
 func (c *Client) loop() {
+	c.timer = time.NewTimer(opTimeout)
+	if !c.timer.Stop() {
+		<-c.timer.C
```

Comment:
why are we waiting for the opTimeout here?

I guess it's because of the following `c.timer.C`? But I feel there should be a better way.

Also, waiting for an expired timer seems bad.

Reply:
The Timer API doesn't allow the creation of a timer without a timeout value set.  I made the Timer and immediately stopped it so the channel that the timer uses would exist.   Another approach could be to create the timer with the maximum Duration.